### DISCUSSION TITLE
Class prop. default value

### DISCUSSION THREAD

**Discussion by SebGue** [Eomys/pyleecan#335] [Q&A]
Hello @BonneelP, 

is it possible to set set a class property of type X to a default value of x. With x is an inherited class of X.
Less abstract: I have created an end_winding prop. on Winding. Prop. type is abstract class EndWinding. Now I would like to set this prop. to EndWindingCirc (inherited by EndWinding) by default.  
How could I do that?

Best regards, Sebastian

**Comment by BonneelP** [comment]
Hello Sebastian,

The idea would be to define the default value of the property as "EndWindingCirc( )" (in the .csv), then in the __init__ the default value would be -1 and in the setter if we have -1 we set EndWindingCirc( ) (to avoid sharing the same object in all instances).
I think that we already have this feature in the class generator but I need to check.
By the way, we opened an issue a long time ago on endwinding #32 so if you plan to share your work on this aspect it could be great to talk about it in this issue :)

Best regards,
Pierre

**Reply by SebGue** [reply]
Hello Pierre,

My fault, I missed the brackets when setting default to EndWindingCirc. Now EndWindingCirc is imported to winding classes but still defaults to EndWinding. So it seems to be implemented partially only. BR Sebastian

**Reply by SebGue** [reply]
OT: Is there also an old issue on vents since I found a but there as well?

**Reply by BonneelP** [reply]
I don't think that we have an issue on ventilation ducts for now.
